### ISSUE TITLE
Rule for promises inside event handlers

### ISSUE THREAD

**Issue by MrTrick** [xjamundx/eslint-plugin-promise#125]
### Description

When code must run multiple times, at unpredictable intervals, in response to user action, then event handling patterns are common and useful:

```javascript
obj.onsomeevent = function(evt) {
 ...
};
//OR
obj.on('someevent', function(evt) {
  ...
});
```

These handler functions might use or interact with promises.  
The event dispatcher seldom examines or cares about whether the handler returns a value.

The `catch-or-return` rule asks users to add a catch() block or return the promise.
Adding a 'return' statement is the easy way out, but an event handler should **not** return a promise.
(And doing so actually *fix* the code - rejections will be ignored)

```javascript
//BAD code that doesn't detect errors.
obj.on('someevent', function(evt) {
   return fetch(url).then(doSomethingImportant);
})
```

### Suggestion

Could a lint rule detect event handler functions, and disallow returning promises from inside them?
By extension, could it also detect this situation:

```javascript
//okay, I suppose?
function doSomethingImportant() {
   return fetch(url).then(thatImportantThing);
}
//BAD code again, just a layer of indirection.
obj.on('someevent', function(evt) {
   return doSomethingImportant();
})
```
